tests/models/test_hooks.py

```diff
@@ -401,19 +420,30 @@ def test_trainer_model_hook_system_fit(tmpdir):
         'pytorch-lightning_version': __version__,
         'state_dict': ANY,
     }
+    if kwargs.get('amp_backend') == 'native':
+        saved_ckpt['native_amp_scaling_state'] = ANY
+    elif kwargs.get('amp_backend') == 'apex':
+        saved_ckpt['amp_scaling_state'] = ANY
+    device = torch.device('cuda:0' if 'gpus' in kwargs else 'cpu')
+
     expected = [
         dict(name='Callback.on_init_start', args=(trainer, )),
         dict(name='Callback.on_init_end', args=(trainer, )),
         dict(name='prepare_data'),
         dict(name='configure_callbacks'),
         dict(name='Callback.on_before_accelerator_backend_setup', args=(trainer, model)),
+        # FIXME
+        *([dict(name='train_dataloader')] if kwargs.get('plugins') == 'deepspeed' else []),
```

Comment:
@SeanNaren This is probably from the call to `train_dataloader()` in `_auto_select_batch_size`. Is that okay?

Reply:
Yep sounds good